aviatrix/resource_aviatrix_copilot_security_group_management_config.go

```diff
@@ -0,0 +1,265 @@
+		Region:                               d.Get("region").(string),
+		Zone:                                 d.Get("zone").(string),
+		VpcId:                                d.Get("vpc_id").(string),
+		InstanceId:                           d.Get("instance_id").(string),
+		EnableCopilotSecurityGroupManagement: d.Get("enable_copilot_security_group_management").(bool),
+	}
+
+	return copilotSecurityGroupManagementConfig
+}
+
+func resourceAviatrixCopilotSecurityGroupManagementConfigCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*goaviatrix.Client)
+
+	copilotSecurityGroupManagementConfig := marshalCopilotSecurityGroupManagementConfigInput(d)
+
+	d.SetId(strings.Replace(client.ControllerIP, ".", "-", -1))
+	flag := false
+	defer resourceAviatrixCopilotSecurityGroupManagementConfigReadIfRequired(ctx, d, meta, &flag)
+
+	if copilotSecurityGroupManagementConfig.EnableCopilotSecurityGroupManagement {
+		if copilotSecurityGroupManagementConfig.CloudType == 0 || copilotSecurityGroupManagementConfig.AccountName == "" ||
+			copilotSecurityGroupManagementConfig.VpcId == "" || copilotSecurityGroupManagementConfig.InstanceId == "" ||
+			(copilotSecurityGroupManagementConfig.Region == "" && copilotSecurityGroupManagementConfig.Zone == "") {
+			return diag.Errorf("'cloud_type', 'account_name', 'region'/'zone', 'vpc_id' and 'instance_id' are required to enable copilot security group management")
+		}
+
+		if goaviatrix.IsCloudType(copilotSecurityGroupManagementConfig.CloudType, goaviatrix.AWSRelatedCloudTypes|goaviatrix.AzureArmRelatedCloudTypes) &&
+			copilotSecurityGroupManagementConfig.Zone != "" {
+			return diag.Errorf("'zone' is only supported for 'GCP', please use 'region' for AWS and Azure")
+		}
+
+		if goaviatrix.IsCloudType(copilotSecurityGroupManagementConfig.CloudType, goaviatrix.GCPRelatedCloudTypes) &&
+			copilotSecurityGroupManagementConfig.Region != "" {
+			return diag.Errorf("'region' is only supported for AWS and Azure, please use 'zone' for GCP")
+		}
+
+		if err := client.EnableCopilotSecurityGroupManagement(ctx, copilotSecurityGroupManagementConfig); err != nil {
+			return diag.Errorf("could not enable copilot security group management: %v", err)
+		}
+	} else {
+		if copilotSecurityGroupManagementConfig.CloudType != 0 || copilotSecurityGroupManagementConfig.AccountName != "" ||
+			copilotSecurityGroupManagementConfig.VpcId != "" || copilotSecurityGroupManagementConfig.InstanceId != "" ||
+			copilotSecurityGroupManagementConfig.Region != "" || copilotSecurityGroupManagementConfig.Zone != "" {
+			return diag.Errorf("'cloud_type', 'account_name', 'region'/'zone', 'vpc_id' and 'instance_id' are not needed to disable copilot security group management")
+		}
+
+		err := client.DisableCopilotSecurityGroupManagement(ctx)
+		if err != nil {
+			if strings.Contains(err.Error(), "not enabled") {
+				log.Printf("copilot security group management is already disabled")
```

Comment:
I feel like it shouldn't be an error to call the disable API when it is already disabled. For Private Mode, I check the error message in the `CheckFunc` and return nil if it is already disabled.